data/src/main/java/com/microsoft/azure/kusto/data/Utils.java

```diff
@@ -211,4 +218,398 @@ public static InputStream resolveInputStream(HttpResponse response, InputStream
+     */
+    public static String prependIfMissing(String str, String prefix) {
+        if (str == null || prefix == null) {
+            return str;
+        }
+        if (!str.startsWith(prefix)) {
+            return prefix + str;
+        }
+        return str;
+    }
+
+    /**
+     * Appends the suffix to the end of the string if the string does not already end with any of the suffixes.
+     *
+     * @param str the string to process
+     * @param suffix the suffix to append to the end of the string
+     * @param suffixes additional suffixes that are valid terminators
+     * @return a new String if suffix was appended, the same string otherwise
+     */
+    public static String appendIfMissing(String str, CharSequence suffix, CharSequence... suffixes) {
+        if (str == null || suffix == null) {
+            return str;
+        }
+        
+        // Check if string already ends with the main suffix
+        if (str.endsWith(suffix.toString())) {
+            return str;
+        }
+        
+        // Check if string ends with any of the additional suffixes
+        if (suffixes != null) {
+            for (CharSequence s : suffixes) {
+                if (s != null && str.endsWith(s.toString())) {
+                    return str;
+                }
+            }
+        }
+        
+        // String doesn't end with any of the suffixes, append the main suffix
+        return str + suffix.toString();
+    }
+
+    /**
+     * Checks if a string is empty (null or zero length).
+     *
+     * @param str the string to check
+     * @return true if the string is null or has zero length
+     */
+    public static boolean isEmpty(String str) {
+        return str == null || str.length() == 0;
+    }
+
+    /**
+     * Case-insensitive check if a CharSequence ends with a specified suffix.
+     * 
+     * nulls are handled without exceptions. Two null references are considered to be equal.
+     * The comparison is case insensitive.
+     *
+     * @param str the CharSequence to check, may be null
+     * @param suffix the suffix to find, may be null
+     * @return true if the CharSequence ends with the suffix, case-insensitive, or both null
+     */
+    public static boolean endsWithIgnoreCase(CharSequence str, CharSequence suffix) {
+        if (str == null || suffix == null) {
+            return str == null && suffix == null;
+        }
+        if (suffix.length() > str.length()) {
+            return false;
+        }
+        return str.toString().regionMatches(true, str.length() - suffix.length(), suffix.toString(), 0, suffix.length());
+    }
+
+    /**
+     * Custom implementation to replace Apache Commons Text TextStringBuilder.
+     * Provides formatted message building capabilities with line appending.
+     * This implementation matches the behavior of TextStringBuilder for the specific
+     * usage pattern in the codebase.
+     */
+    public static class MessageBuilder {
```

Comment:
Do we really need this class?
We literally use it once in the code, and all it does is saves us a call to String.Format?